cumulusci/salesforce_api/org_schema.py

```diff
@@ -229,27 +314,59 @@ def close(self) -> None:
 
 @contextmanager
 def get_org_schema(
-    sf: Salesforce,
-    org_config: OrgConfig,
-    include: T.Sequence[str] = None,
+    sf,
+    org_config,
     *,
+    include_counts: bool = False,
+    filters: T.Sequence[Filters] = (),
+    patterns_to_ignore: T.Tuple[str] = (),
+    included_objects: T.List[str] = (),
     force_recache=False,
     logger=None,
 ):
     """
     Get a read-only representation of an org's schema.
-
+    sf - simple_saleforce object
     org_config - an OrgConfig for the relevant org
+
+    include_counts: query each queryable/retrievable object count.
+                    This takes time and may even timeout in huge orgs!
+    filters: A sequence of Filters which are the same as Salesforce SObject properties
+            like .createable, .deletable etc. Objects that do not match are ignored.
+            Two special filters exist:
+                * Filters.extractable, which uses heuristics to limit to objects
+                    that extract properly
+                * Filters.populated, which limits to objects that have data in them.
+                    This depends upon `include_counts`
+    included_objects: Ignore objects not in this list. Stacks with other filters
+    patterns_to_ignore: Strings in SQL %LIKE% syntax that match SObjects to be
+                        ignored.
+
     force_recache: True - replace cache. False (default) - use/update cache is available.
     logger - replace the standard logger "cumulusci.salesforce_api.org_schema"
     """
+    assert not isinstance(patterns_to_ignore, str)
+
+    filters = set(filters)
     with org_config.get_orginfo_cache_dir(Schema.__module__) as directory:
         directory.mkdir(exist_ok=True, parents=True)
         schema_path = directory / "org_schema.db.gz"
 
         if force_recache and schema_path.exists():
             schema_path.unlink()
 
+        if Filters.populated in filters:
+            filters.add(Filters.queryable)
+            filters.add(Filters.retrieveable)
+            # experiment with removing this limitation by using limit 1 query instead
+            assert include_counts, "Filters.populated depends on include_counts"
+            patterns_to_ignore += NOT_COUNTABLE
+
+        if Filters.extractable in filters:
```

Comment:
Doesn't need to be in version 1, but we might want to add a "loadable" criterion (`queryable` + `retrieveable` + `createable`)